js-plugin/src/main/kotlin/org/jetbrains/research/deepbugs/javascript/ide/inspections/common/JSDeepBugsBinOperatorInspection.kt

```diff
@@ -0,0 +1,47 @@
+package org.jetbrains.research.deepbugs.javascript.ide.inspections.common
+
+import com.intellij.codeInsight.lookup.LookupElementBuilder
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.lang.javascript.psi.JSBinaryExpression
+import com.intellij.psi.PsiElement
+import org.jetbrains.research.deepbugs.common.datatypes.BinOp
+import org.jetbrains.research.deepbugs.common.datatypes.DataType
+import org.jetbrains.research.deepbugs.common.ide.problem.BugDescriptor
+import org.jetbrains.research.deepbugs.common.ide.quickfixes.ReplaceBinOperatorQuickFix
+import org.jetbrains.research.deepbugs.common.model.CommonModelStorage
+import org.jetbrains.research.deepbugs.javascript.JSDeepBugsConfig
+import org.jetbrains.research.deepbugs.javascript.JSResourceBundle
+import org.jetbrains.research.deepbugs.javascript.ide.inspections.base.JSDeepBugsBinExprInspection
+import org.jetbrains.research.deepbugs.javascript.ide.quickfixes.JSIgnoreExpressionQuickFix
+import org.jetbrains.research.deepbugs.javascript.ide.quickfixes.utils.operators
+import org.jetbrains.research.keras.runner.nn.model.sequential.Perceptron
+
+class JSDeepBugsBinOperatorInspection : JSDeepBugsBinExprInspection() {
+    override val model: Perceptron?
+        get() = CommonModelStorage.common.binOperatorModel
+
+    override fun createProblemDescriptor(node: PsiElement, data: DataType): ProblemDescriptor {
+        val textRange = (node as JSBinaryExpression).operationNode!!.textRange
```

Comment:
If input type of node is defined by class -- why not to make it generic parameter?